### ISSUE TITLE
Fix S1645 FP: Not applicable for XName/XNamespace

### ISSUE THREAD

**Issue by PavlinII** [SonarSource/sonar-dotnet#4346]
### Description

VB.NET S1645 raises FP when `XName` is constructed from `XNamespace` and string.

### Repro steps

```vb
        Private Shared Sub ParseMsg(Root As XElement)
            Dim NS As XNamespace = Root.Name.Namespace
            Dim Msgs As XElement = Root.Element(NS + "runtimeMsgs")  ' FP: Switch this use of the '+' operator to the '&'
        End Sub
```

### Expected behavior

Operator `&` is not defined for `XNamespace` and `string` => it should not raise the issue.

### Related information

* SonarLint 4.33.1
* Visual Studio version 2019
